Firestore/core/src/firebase/firestore/remote/serializer.cc

```diff
@@ -291,7 +336,7 @@ void EncodeNestedFieldValue(pb_ostream_t* stream,
   size_t size = substream.bytes_written;
 
   // Write out the size to the output stream.
-  EncodeVarint(stream, size);
+  PbOstream(stream).EncodeSize(size);
```

Comment:
This may just be a hot take on my part, but it seems that your abstraction over `pb_ostream_t` is incomplete. What if instead of taking a paramter of `pb_ostream_t*` you took a `PbOstream*` (or whatever you call it)?